src/Interpreters/CrossToInnerJoinVisitor.cpp

```diff
@@ -275,7 +280,20 @@ bool getTables(ASTSelectQuery & select, std::vector<JoinedElement> & joined_tabl
     }
 
     if (num_using && (num_tables - num_array_join) > 2)
-        throw Exception("Multiple CROSS/COMMA JOIN do not support USING", ErrorCodes::NOT_IMPLEMENTED);
+    {
+        std::stringstream ss;
```

Comment:
It's better to extract this logic to own function. Just to ignore it in general code reading. Something like.
```
throw Exception(createErrorMessage(tables_with_using), ErrorCodes::NOT_IMPLEMENTED);
```

Reply:
Made a function for printing arrays of AST pointers.